### ISSUE TITLE
Intellij stuck updating cargo.io index

### ISSUE THREAD

**Issue by Ragarnoy** [intellij-rust/intellij-rust#10214]
## Environment

* **IntelliJ Rust plugin version: 0.4.189.5234-223**
* **Rust toolchain version: 1.67.1**
* **IDE name and version: Clion 2022.3.2**
* **Operating system: Windows 11**

## Problem description
I don't have any linter, the plugin simply does not offer any help, the bottom bar is stuck "Updating crates.io index" and nothing happens  and the IDE doesn't detect any rust crate.

I tried invalidating the cache, repairing the IDE, uninstalling it, downgrading it, nothing seems to fix it.

 ## Steps to reproduce
1. Open any rust project
2. wait

**Comment by Ragarnoy** [comment]
this was caused by having a wsl toolchain but clion not being able to detect it, causing this error in the logs:

```
2023-03-07 15:05:12,844 [ 612161]   INFO - #c.j.c.cpp - Cannot read GDB version: 
com.intellij.execution.ExecutionException: WSL distribution 'Ubuntu' is not found
	at com.jetbrains.cidr.cpp.toolchains.WSLCidrUtil$WSLHost$1.build(WSLCidrUtil.java:156)
	at com.jetbrains.cidr.system.CidrProcessBuilder.build(CidrProcessBuilder.kt:41)
	at com.jetbrains.cidr.system.HostMachine.runProcess(HostMachine.java:58)
	at com.jetbrains.cidr.system.HostMachine.runProcess(HostMachine.java:48)
	at com.jetbrains.cidr.cpp.toolchains.GDB.getGdbVersionOutput(GDB.java:81)
	at com.jetbrains.cidr.cpp.toolchains.GDB.readToolVersion(GDB.java:54)
	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainUsageCollector$Impl.doWork(CPPToolchainUsageCollector.java:86)
	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainUsageCollector.getMetrics(CPPToolchainUsageCollector.java:65)
	at com.intellij.internal.statistic.service.fus.collectors.ApplicationUsagesCollector.getMetricsAsync$suspendImpl(ApplicationUsagesCollector.kt:48)
	at com.intellij.internal.statistic.service.fus.collectors.ApplicationUsagesCollector.getMetricsAsync(ApplicationUsagesCollector.kt)
	at com.intellij.internal.statistic.service.fus.collectors.FUStateUsagesLogger$logApplicationStates$3$1.invokeSuspend(FUStateUsagesLogger.kt:192)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

and for some reason preventing rust intellij from working somehow
